### DISCUSSION TITLE
How to obtain the value of --data

### DISCUSSION THREAD

**Discussion by zhangwuqiao** [actix/actix-web#3013] [Q&A] [answered]
```sh
curl -X GET -H 'token: ck-e15c4fe5-b9b5-4153-ba35-af22182592b2' \
  -H 'Content-Type: text/plain' -i 'http://127.0.0.1:8080/index2' \
  --data '"test":[{"a":"str","b":12},{"a":"str1","b":121},{"a":"str2","b":122}]'
```

How to use an extractor to extract the value of “--data”?

thank you.

**Comment by robjtede** [comment, marked as answer]
It looks like you're trying to send JSON, however that snippet isn't valid JSON.

Using the [`Json<T>`](https://actix.rs/docs/extractors/#json) extractor ([docs](https://docs.rs/actix-web/4/actix_web/web/struct.Json.html)) you would be able to obtain a struct with the sent data.

Alternatively, use `String` or `Bytes` as the extractor to get the raw string or byte payload respectively.

**Reply by robjtede** [reply]
I also noticed that you're sending GET request. It isn't usually valid to send a body here, so I'd recommend switching it to a POST.
